src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs

```diff
@@ -969,5 +1031,20 @@ public static implicit operator MergedDatabaseObservationState(DatabaseObservati
                 return new MergedDatabaseObservationState(state.RawDatabase, state);
             }
         }
+
+        internal TestingStuff ForTestingPurposes;
+
+        internal TestingStuff ForTestingPurposesOnly()
+        {
+            if (ForTestingPurposes != null)
+                return ForTestingPurposes;
+
+            return ForTestingPurposes = new TestingStuff();
+        }
+
+        internal sealed class TestingStuff
+        {
+            internal StringBuilder DiagnosticLogBuilder;
```

Comment:
change to `Action<string>` and rename to e.g. `OnDiagnosticLog` or sth like that

Reply:
Dםמק